pkg/resource/claim_binding_reconciler.go

```diff
@@ -138,6 +138,20 @@ type ManagedFinalizer interface {
 	Finalize(ctx context.Context, cm Managed) error
 }
 
+// A FinalizerChain chains multiple managed finalizers.
+type FinalizerChain []ManagedFinalizer
```

Comment:
Noticed it the first time in this PR but why do we call `r.managed.Finalize` in claim reconciler? Shouldn't claim reconciler just send the deletion signal and checks whether object is deleted, i.e `(NotFound)` and then calls its claim's finalizer? Pretty much like how managed reconciler treats its external resource.

Reply:
In both of these reconcilers the "Finalize" interface is not intended explicitly to mean "remove a finalizer" in the metadata sense, but rather "finalize the lifecycle of this object". I can see how this is confusing, but struggled to find a better name for the interface.

The managed resource reconciler finalizes its work on a managed resource by removing its finalizer. The claim binding reconclise finalizes its work on a managed resource by removing its claim ref and setting it to unbound.